python/mxnet/_deferred_compute.py

```diff
@@ -0,0 +1,95 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+"""Deferred Compute for NDArray."""
+
+import ctypes
+import contextlib
+
+from .base import _LIB, check_call, SymbolHandle, _as_list
+from .symbol import Symbol
+
+__all__ = []
+
+def is_deferred_compute():
+    """Get status of deferred compute mode."""
+    curr = ctypes.c_bool()
+    check_call(_LIB.MXNDArrayIsDeferredComputeEnabled(ctypes.byref(curr)))
+    return curr.value
+
+def set_deferred_compute(is_deferred_compute):
+    """Enable / Disable deferred compute mode.
+
+    Parameters
+    ----------
+    is_deferred_compute: bool
+
+    Returns
+    -------
+    Previous deferred compute state.
+    """
+    prev = ctypes.c_int()
+    check_call(_LIB.MXNDArraySetDeferredComputeEnabled(
+        ctypes.c_int(is_deferred_compute), ctypes.byref(prev)))
+    return bool(prev.value)
+
+
+@contextlib.contextmanager
+def context():
+    # Like other MXNet context manager, this bleeds state across concurrent
```

Comment:
Can you create a github issue to trace it, or mention potential workarounds?

Reply:
does this have any issues with the work in #16654 ?

Reply:
This refers to problems if users use https://docs.python.org/3/library/asyncio.html